app/src/main/java/org/schabi/newpipe/streams/Mp4DashReader.java

```diff
@@ -317,7 +313,7 @@ private String boxName(int type) {
     private Box readBox() throws IOException {
         Box b = new Box();
         b.offset = stream.position();
-        b.size = stream.readInt();
+        b.size = stream.readUnsignedInt();

```

Comment:
Was it a bug that it had `readInt()` instead of `readUnsignedInt()` here? Or did you somehow accidentally change this? Other than this, this PR looks good, but I haven't tested it yet.

Reply:
no, the fix is here https://github.com/TeamNewPipe/NewPipe/pull/3325/commits/62d934dd8e89cc751c90e0c6a9ec36ea606cfbee

Reply:
I understand that, but why did you change this line?

Reply:
* in somewhere was an overflow (the file was to big), so i discarded every possible overflow. Also, in mp4 box sizes are unsigned values, but since in DASH streams, bigger boxes llike `mdat` never reachs 2GB
* yes, size cannot be negative